src/BLEMIDI_Transport.h

```diff
@@ -229,108 +234,158 @@ class BLEMIDI_Transport
                     }
                     break;
-                case MIDI_NAMESPACE::MidiType::ProgramChange:
-                case MIDI_NAMESPACE::MidiType::AfterTouchChannel:
+                case 0xC0:
+                case 0xD0:
+#ifdef RUNNING_ENABLE
+                    mBleClass.add(lastStatus);
+#endif
                     for (auto i = lPtr; i < rPtr; i = i + 1)
                     {
+#ifndef RUNNING_ENABLE
                         mBleClass.add(lastStatus);
+#endif
                         mBleClass.add(buffer[i + 1]);
                     }
                     break;
-                case MIDI_NAMESPACE::MidiType::SystemExclusive:
-
-                    mBleClass.add(buffer[lPtr]);
+                case 0xF0:
+                    mBleClass.add(lastStatus);
                     for (auto i = lPtr; i < rPtr; i++)
                         mBleClass.add(buffer[i + 1]);
 
                     break;
+
+                default:
+                    break;
+                }
+            }
+            else
+            {
+#ifndef RUNNING_ENABLE
+                auto midiType = lastStatus & 0xF0;
+                if (sysExContinuation)
+                    midiType = 0xF0;
+
+                switch (midiType)
+                {
+                case 0x80:
+                case 0x90:
+                case 0xA0:
+                case 0xB0:
+                case 0xE0:
+                    //3 bytes full Midi -> 2 bytes runningStatus
+                    for (auto i = lPtr; i <= rPtr; i = i + 2)
+                    {
+                        mBleClass.add(lastStatus);
+                        mBleClass.add(buffer[i]);
+                        mBleClass.add(buffer[i + 1]);
+                    }
+                    break;
+                case 0xC0:
+                case 0xD0:
+                    //2 bytes full Midi -> 1 byte runningStatus
+                    for (auto i = lPtr; i <= rPtr; i = i + 1)
+                    {
+                        mBleClass.add(lastStatus);
+                        mBleClass.add(buffer[i]);
+                    }
+                    break;
+
                 default:
                     break;
                 }
+#else
+                mBleClass.add(lastStatus);
+                for (auto i = lPtr; i <= rPtr; i++)
+                    mBleClass.add(buffer[i]);
+#endif
+                runningStatusContinuation = false;
             }
 
             if (++rPtr >= length)
                 return; // end of packet
 
+            if (lastStatus < 0xf0) //exclude System Message. They must not be RunningStatus
+            {
+                previousStatus = lastStatus;
+            }
+
             timestampByte = buffer[rPtr++];
             if (timestampByte >= 80) // is bit 7 set?
             {
-                timestamp = setMidiTimestamp(headerByte, timestampByte); 
-                // what do to with the timestamp?
+                auto timestampLow = 0x7f & timestampByte;
```

Comment:
use the setMidiTimestamp in `BLEMIDI_Transport.h` (not avail. in the msvc test app)